otnstester/OtnsTest.go

```diff
@@ -0,0 +1,365 @@
+package otnstester
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	visualize_grpc_pb "github.com/openthread/ot-ns/visualize/grpc/pb"
+	"google.golang.org/grpc"
+
+	"github.com/openthread/ot-ns/dispatcher"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/chzyer/readline"
+	"github.com/openthread/ot-ns/cli/runcli"
+	"github.com/openthread/ot-ns/otns_main"
+	"github.com/openthread/ot-ns/progctx"
+	. "github.com/openthread/ot-ns/types"
+	"github.com/openthread/ot-ns/visualize"
+	"github.com/simonlingoogle/go-simplelogger"
+)
+
+var (
+	stdinPipeFile  = "stdin.namedpipe"
+	stdoutPipeFile = "stdout.namedpipe"
+)
+
+type OtnsTest struct {
+	*testing.T
+
+	stdin                  *os.File
+	stdout                 *os.File
+	otnsDone               chan struct{}
+	pendingOutput          chan string
+	pendingVisualizeEvents chan *visualize_grpc_pb.VisualizeEvent
+	stdinCloser            *readline.CancelableStdin
+	ctx                    *progctx.ProgCtx
+	grpcClient             visualize_grpc_pb.VisualizeGrpcServiceClient
+	visualizeStream        visualize_grpc_pb.VisualizeGrpcService_VisualizeClient
+}
+
+func (ot *OtnsTest) Go(duration time.Duration) {
+	seconds := float64(duration) / float64(time.Second)
+	ot.sendCommand(fmt.Sprintf("go %f", seconds))
+	ot.expectDone()
+}
+
+func (ot *OtnsTest) Join() {
```

Comment:
nit: the name `ot` is misleading :(

Reply:
it is, but can't think of a better name.